core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -94,14 +95,14 @@
     /**
      * Possibly null proxy user name.
      */
-    private final String userName;
+    private String userName;
 
     /**
      * List of host names that shouldn't use proxy, as typed by users.
      *
      * @see #getNoProxyHostPatterns()
      */
-    public final String noProxyHost;
+    public String noProxyHost;
```

Comment:
It immediately makes it a public API. It should be at least annotated with `@Restricted(NoExternalUse.class)`